### DISCUSSION TITLE
Stacked Improvements

### DISCUSSION THREAD

**Discussion by FilledStacks** [FilledStacks/stacked#210] [Ideas]
Stacked is growing! Which means more questions and better indications of what's required to build maintainable applications. I wanted to open this discussion to hear what kind of improvements you'd like to see in the stacked package. You can suggest anything, prioritising a bug that affects you, asking for a new feature, improvement in naming or code style. All are welcome. 

The two things that has been asked the most is:

1. How do we handle forms in Stacked
2. How do I communicate back to the UI from the VIewModel

Because of that the changes / implementation that I'd like to tackle first are:

1. Building a FormViewModel with a definitive guide for building forms using Stacked
2. A dedicated pattern to use for communicating back to the ViewModel supported in the `ViewModelBuilder`.

In addition to these things above there is [this issue](https://github.com/FilledStacks/stacked/issues/171) that I'd like to solve as well as [this one](https://github.com/FilledStacks/stacked/issues/199). 

I am hoping to dedicate more of our companies resources at FilledStacks to helping me in the open source efforts but for the time being there's no hard dates on when this will happen. At the minimum I will start building it when it's required in the new BoxtOut applications. 

Please leave questions, suggestions, anything below and lets talk about what you'd like to see in this package!

**Comment by d3tr0id** [comment]
Hi there!
Just wanted to check if I understood correctly: Point 2. `A dedicated pattern to use for communicating back to the ViewModel supported in the ViewModelBuilder.` By this, you mean that if I would have a ReactiveService that notifies my FooBarViewModel (ReactiveViewModel) this can be used to call a function of my ViewModel from within the related View since it will be rebuilt?

Thanks in advance!
